smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java

```diff
@@ -0,0 +1,336 @@
+
+    private static final String REST_FILESYSTEM = "opennms/rest/filesystem";
+    private static final String FILE_NAME = "pom.xml";
+    private static final String USERNAME = "editor";
+    private static final String PASSWORD = "admin";
+
+    @ClassRule
+    public static final OpenNMSStack STACK = OpenNMSStack.MINIMAL;
+
+    @Before
+    public void setUp() throws InterruptedException {
+
+        RestAssured.baseURI = STACK.opennms().getBaseUrlExternal().toString();
+        RestAssured.port = STACK.opennms().getWebPort();
+        RestAssured.authentication = preemptive().basic(BASIC_AUTH_USERNAME, BASIC_AUTH_PASSWORD);
+
+        addUserAPI();
+
+        Thread.sleep(5000);
+    }
+
+    /**
+     * Test to verify that the normal behaviour will work
+     */
+    @Test
+    public void normalFlow() {
+        LOG.info("Normal flow test");
+        RequestBody body;
+
+        // upload a new file
+        try {
+            File file = new File(FILE_NAME);
+            body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                    .addFormDataPart("upload", FILE_NAME,
+                            RequestBody.create(MediaType.parse("application/octet-stream"),
+                                    file))
+                    .build();
+
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "POST", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // update the file
+        body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                .addFormDataPart("upload", "<name>OpenNMS Smoke Test</name>")
+                .build();
+
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "POST", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // get file
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "GET", null);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+
+        // remove the file
+        MediaType mediaType = MediaType.parse("text/plain");
+        body = RequestBody.create(mediaType, "");
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "DELETE", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Test that the File Editor will deny file that is not in the correct format
+     */
+    @Test
+    public void wrongFileName() {
```

Comment:
May want a more specific method name (and log message and comment), for example `unsupportedFileExtension`, comment: "Test that the File Editor will deny a file with an unsupported file extension".